tests/unit/src/test/scala/tests/codeactions/ExtractValueLspSuite.scala

```diff
@@ -0,0 +1,159 @@
+package tests.codeactions
+
+import scala.meta.internal.metals.codeactions.ExtractValueCodeAction
+
+class ExtractValueLspSuite
+    extends BaseCodeActionLspSuite("extractValueRewrite") {
+
+  check(
+    "block",
+    """|object Main {
+       |  def method2(i: Int) = ???
+       |  def method1(s: String): Unit = {
+       |    method2(1 + <<2>>)
+       |  }
+       |
+       |}
+       |""".stripMargin,
+    ExtractValueCodeAction.title,
+    """|object Main {
+       |  def method2(i: Int) = ???
+       |  def method1(s: String): Unit = {
+       |    val newValue = 1 + 2
+       |    method2(newValue)
+       |  }
+       |
+       |}
+       |""".stripMargin
+  )
+
+  check(
+    "for-comprehension",
+    """|object Main {
+       |  def method2(i: Int) = ???
+       |  
+       |  def main() = {
+       |    val opt = Option(1)
+       |    for {
+       |       i <- opt
+       |       res = method2(i + 23 + <<123>>)      
```

Comment:
How would it behave if this was `res <- method2(i + 23 + <<123>>)` and it was the first line in a `for`?

Reply:
It will bug out! Good catch :sweat_smile:

Reply:
Aaaand it's fixed!